OpenRA.Mods.Common/ServerTraits/LobbyCommands.cs

```diff
@@ -135,6 +135,10 @@ public bool InterpretCommand(S server, Connection conn, Session.Client client, s
 
 						client.Slot = s;
 						S.SyncClientToPlayerReference(client, server.MapPlayers.Players[s]);
+
+						var validatedColor = ColorValidator.ValidatePlayerColorAndGetAlternative(server, client.Color, client.Index, conn);
+						client.PreferredColor = client.Color = validatedColor;
```

Comment:
If I don't set it up the color is invalid every time you change slots, so everyone in the lobby get a message that some color is invalid, without even being able to see what and why. Also you get a new color every time you change slots.
Setting PreferredColor means that notification occurs only the first time you change slots.